File: problems/HackerRank/twoArrays.ts
function twoArrays(k: number, A: number[], B: number[]): string {
  A.sort((a, b) => a - b);
  B.sort((a, b) => b - a);

  for (let i = 0; i < A.length; i++) {
    const sum = A[i] + B[i];
    if (sum < k) {
      return 'NO';
    }
  }

  return 'YES';
}

console.log(twoArrays(5, [1, 2, 3, 4, 3], [3, 5, 1, 2, 9, 0]));

File: problems/HackerRank/week3/twoArrays.py
def twoArrays(k, A, B):
    if len(A) != len(B): 
        return 'NO'

    A.sort()
    B.sort()
    B.reverse()
    
    for i in range(len(A)):
        if A[i] + B[i] < k:
            return 'NO'
    
    return 'YES'